mobile/application-manager-base.ts

```diff
@@ -94,7 +97,43 @@ export abstract class ApplicationManagerBase extends EventEmitter implements Mob
 			this.lastAvailableDebuggableApps = currentlyAvailableDebuggableApps;
 
 			_.each(newAvailableDebuggableApps, (appInfo: Mobile.IDeviceApplicationInformation) => this.emit("debuggableAppFound", appInfo));
-			_.each(notAvailableAppsForDebugging, (appInfo: Mobile.IDeviceApplicationInformation) => this.emit("debuggableAppLost", appInfo));
+			_.each(notAvailableAppsForDebugging, (appInfo: Mobile.IDeviceApplicationInformation) => {
+				this.emit("debuggableAppLost", appInfo);
+
+				if (_.has(this.lastAvailableDebuggableAppViews, appInfo.appIdentifier)) {
+					// Prevent emitting debuggableViewLost when application cannot be debugged anymore.
+					delete this.lastAvailableDebuggableAppViews[appInfo.appIdentifier];
+				}
+			});
+
+			let cordovaDebuggableAppIdentifiers = _(currentlyAvailableDebuggableApps)
+				.filter(c => c.framework === TARGET_FRAMEWORK_IDENTIFIERS.Cordova)
+				.map(c => c.appIdentifier)
+				.value();
+
+			let currentlyAvailableAppViews = this.getDebuggableAppViews(cordovaDebuggableAppIdentifiers).wait();
+
+			_.each(currentlyAvailableAppViews, (currentlyAvailableViews, appIdentifier) => {
+				let previouslyAvailableViews = this.lastAvailableDebuggableAppViews[appIdentifier];
+
+				let newAvailableViews = _.differenceBy(currentlyAvailableViews, previouslyAvailableViews, "id");
+				let notAvailableViews = _.differenceBy(previouslyAvailableViews, currentlyAvailableViews, "id");
+
+				_.each(notAvailableViews, debugWebViewInfo => this.emit("debuggableViewLost", appIdentifier, debugWebViewInfo));
```

Comment:
I'd prefer if this lambda expression did not return the result of `this.emit` because we might get unexpected results in the event of `this.emit` returning `false` or `undefined`